### PR TITLE
Made a start on Dashboard functionality.

### PR DESCRIPTION
**Pull request by nmcmahon-unosquare**
- Implemented get logged in employee's events for a given month.
- Add service to Dependency injection.
- Added functionality to get logged in user from Employee Credentials
- Changed EventService to Event object is populated with Event dates.

### CODE REVIEW

File: v2/AdminCore.Services/DashboardService.cs [hunk 1]
```diff
@@ -3,19 +3,32 @@
...
     {
       throw new NotImplementedException();
     }
 
-    public IList<EventDto> GetEmployeeEvents(int employeeId, DateTime date)
+    public IList<EventDto> GetEmployeeEventsForAGivenMonth(int employeeId, DateTime date)
     {
-      throw new NotImplementedException();
+      var eventsForEmployee = _eventService.GetEventsByEmployeeId(employeeId);
+      return eventsForEmployee.Where(employeeEvent => EventHappensThisMonth(employeeEvent, date)).ToList();
```
Comment (BrownMark) [inline v2/AdminCore.Services/DashboardService.cs]: So here we’re Retrieving all events for an employee and then delimiting in memory, it should be done at the database level

File: v2/AdminCore.Services/DashboardService.cs [hunk 2]
```diff
@@ -36,5 +49,26 @@ public IList<EventDto> GetTeamDashboardEvents(int employeeId, DateTime date)
...
     }
+
+    private static bool EventHappensThisMonth(EventDto employeeEvent, DateTime date)
+    {
+      return employeeEvent.EventDates.Any(eventDate => DateRangeFallsWithinMonth(eventDate.StartDate, eventDate.EndDate, date));
+    }
+
+    private static bool DateRangeFallsWithinMonth(DateTime startDate, DateTime endDate, DateTime givenDate)
+    {
+      var date = startDate.AddDays(0);
```
Comment (BrownMark) [inline v2/AdminCore.Services/DashboardService.cs]: Let’s grab some time of this, I know a better approach.

**Review by BrownMark — CHANGES_REQUESTED** [review]

File: v2/AdminCore.Common/Interfaces/IDashboardService.cs [hunk 3]
```diff
@@ -9,7 +9,7 @@ public interface IDashboardService
   {
     IList<EventDto> GetEmployeeDashboardEvents(int employeeId, DateTime date);
 
-    IList<EventDto> GetEmployeeEvents(int employeeId, DateTime date);
+    IList<EventDto> GetEmployeeEventsForAGivenMonth(int employeeId, DateTime date);
```
Comment (BrownMark) [inline v2/AdminCore.Common/Interfaces/IDashboardService.cs]: I'm not so sure about these naming conventions, perhaps a bit too long winded. Can we do something like 'GetEmployeeEventsForMonth'

**Review by BrownMark — CHANGES_REQUESTED** [review]

File: v2/AdminCore.Services/DashboardService.cs (hunk 1 above)
Comment (nmcmahon-unosquare) [reply v2/AdminCore.Services/DashboardService.cs]: Fixed

File: v2/AdminCore.Common/Interfaces/IDashboardService.cs (hunk 3 above)
Comment (nmcmahon-unosquare) [reply v2/AdminCore.Common/Interfaces/IDashboardService.cs]: Changed to GetEmployeeEventsForMonth

**Pull request closed without merging**